Refund Details: Product & Qty section Mark II

mindgraffiti opened the pull request:
Part 2. Closes #1681.

In this PR the Product & Qty section has the refund summary cell added to it.

For reference, here is the full Refund Details screen:
<img width="410" alt="Screen Shot 2020-01-02 at 2 47 17 PM" src="https://user-images.githubusercontent.com/1062444/71692247-f6e3d400-2d6e-11ea-9152-dbd149d2b955.png">

## To Test
1. On the website, choose an order and create a refund by item. Refund 2 items or more.
2. After creating the refund, go to the app and navigate to the Order Details for the chosen order.
3. The "Refunded" cell should be the new semantic color, "link", which is an active (pink) color. Tap on the "Refunded" cell.
4. The Refund Details screen should appear. As long as the refund was "refund by item", a subtotal, tax, and Products Refund cell will display.

### Screenshots
| Order Details | Refund Details |
| --- | --- |
| <img src="https://user-images.githubusercontent.com/1062444/71732030-d9b11300-2e0b-11ea-8993-b9a274de752c.png" width="350" /> | <img src="https://user-images.githubusercontent.com/1062444/71731924-935bb400-2e0b-11ea-94e9-bfd312b27593.png" width="350" /> |




Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

mindgraffiti committed:
Helper method to return the absolute value of an NSDecimalNumber value

mindgraffiti committed:
Extend the payment table view cell for refund summaries

mindgraffiti committed:
Bugfix: look up the correct refund when more than one exists

mindgraffiti committed:
Add the refund summary cell to Refund Details screen

mindgraffiti committed:
Merge branch 'develop' into issue/1681-product-qty-ii

mindgraffiti committed:
Delete whitespace

pmusolino commented on WooCommerce/Classes/ViewModels/RefundDetails/RefundDetailsViewModel.swift:
```diff
@@ -40,6 +73,41 @@ final class RefundDetailsViewModel {
+    /// Calculate the subtotal for each returned item
+    ///
+    func calculateItemSubtotal() -> NSDecimalNumber {
+        let itemSubtotals = refund.items.map { NSDecimalNumber(string: $0.subtotal) }
```
I think that this NSDecimalNumber allocation can be dangerous since I discovered that NSDecimalNumber uses by default the locale decimal separator to evaluate a decimal amount.

For example if an item price is `10,99` with a comma separator, and in your locale decimal numbers are separated by a point, you will get a new decimal number without decimal: `10.00`. 

I discovered this in this PR where I obtained a wrong formatted amount. https://github.com/woocommerce/woocommerce-ios/pull/1698/files#diff-9602fa47bdc00e54d6c27795709e94acR114
I believe this can also affect the calculation of the total amount.

mindgraffiti replied:
Calculating currency is so tricky 🙂
Thank you for bringing this to my attention. Fixing now!

mindgraffiti replied:
✅

pmusolino commented on WooCommerce/Classes/ViewModels/RefundDetails/RefundDetailsViewModel.swift:
```diff
@@ -40,6 +73,41 @@ final class RefundDetailsViewModel {
+    /// Calculate the subtotal tax for each returned item
+    ///
+    func calculateSubtotalTax() -> NSDecimalNumber {
+        let itemsSubtotalTax = refund.items.map { NSDecimalNumber(string: $0.subtotalTax) }
```
Like above

mindgraffiti replied:
✅

pmusolino reviewed (approved):
Code ✅  just a nit on `NSDecimalNumber` which can affect the subtotal/total amount showed.
Testing ✅ 

| Order Details             |  Refund Details |
:-------------------------:|:-------------------------:
![Simulator Screen Shot - iPhone 11 Pro - 2020-01-06 at 12 45 44](https://user-images.githubusercontent.com/495617/71817738-2a807000-3087-11ea-9bde-7e687ad9e0c1.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-01-06 at 12 46 13](https://user-images.githubusercontent.com/495617/71817782-4edc4c80-3087-11ea-8eb0-e0a3e0baa46f.png)

mindgraffiti committed:
use currencyFormatter's convertToDecimal method

instead of NSDecimalNumber( from string:), because the `convertToDecimal` handles localization for the decimal separator
